doc/time-series.rst

```diff
@@ -133,17 +133,45 @@ For example, we can downsample our dataset from hourly to 6-hourly:
 
 .. ipython:: python
 
-    ds.resample('6H', dim='time', how='mean')
+    ds.resample(time='6H')
 
-Resample also works for upsampling, in which case intervals without any
-values are marked by ``NaN``:
+This will create a specialized ``GroupBy`` object which saves information
+necessary for resampling. All of the reduction methods which work with
+``GroupBy`` objects can also be used for resampling:
 
 .. ipython:: python
 
-    ds.resample('30Min', 'time')
+   ds.resample(time='6H').mean()
+
+You can also supply an arbitrary reduction function to aggregate over each
+resampling group:
+
+.. ipython: python
+
+   ds.resample(time='6H').reduce(np.mean)
+
+Resampling does not yet work for upsampling.
```

Comment:
I think we need to fix this before merging this PR, since it suggests the existing functionality would only exist in deprecated form. Pandas does this with a method called `.asfreq`, though this is basically pure sugar since in practice I think it works exactly the same as `.first` (or `.mean` if only doing pure upsampling).